File: CPP/Program/Vehicle.cpp
//menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
//masukkan library yang digunakan
#include <iostream>
#include <string>

//deklarasi namespace
using namespace std;

// kelas Vehicle yang digunakan untuk merepresentasikan data Vehicle, 
// Vehicle merupakan kelas induk dari kelas SivitasAkademik dan buyut 
// dari kelas Mahasiswa dan Dosen

class Vehicle {
protected:
    // Atribut yang digunakan pada kelas Vehicle terdiri 
    // dari platNomor, mer, tahunProduksi, warna
    string platNomor;
    string merk;
    string tahunProduksi;
    string warna;

public:
    //contructor kosong
    Vehicle(){

    }

    //constructor yg diisi parameter
    Vehicle(string platNomor, string merk, string tahunProduksi, string warna){
        this->platNomor = platNomor;
        this->merk = merk;
        this->tahunProduksi = tahunProduksi;
        this->warna = warna;
    }

    //membuat getter setter
    //set plat nomor
    void setPlatNomor(string platNomor) {
        this->platNomor = platNomor;
    }

    //set merk
    void setMerk(string merk) {
        this->merk = merk;
    }

    //set tahun produksi
    void setTahunProduksi(string tahunProduksi) {
        this->tahunProduksi = tahunProduksi;
    }

    //set warna
    void setWarna(string warna) {
        this->warna = warna;
    }

    //get plat nomor
    string getPlatNomor() {
        return platNomor;
    }

    //get merk
    string getMerk() {
        return merk;
    }

    //get tahun produksi
    string getTahunProduksi() {
        return tahunProduksi;
    }

    //get warna
    string getWarna() {
        return warna;
    }

    //membuat fungsi untuk menampilkan atribut atribut yang ada di kelas vehicle
    virtual void display() const {
        cout << "Plat Nomor: " << platNomor << endl; //nampilin plat nomor
        cout << "Merk: " << merk << endl;   //nampilin merk
        cout << "Tahun Produksi: " << tahunProduksi << endl;    //nampilin tahun produksi
        cout << "Warna: " << warna << endl; //nampilin warna
    }
    
    //destructor
    ~Vehicle() {

    }
};

File: Python/Program/Vehicle.py
class Vehicle:  #membuat kelas induk Vehicle
    def __init__(self, plat_nomor="", merk="", tahun_produksi="", warna=""):  #inisialisasi atribut kendaraan
        self.plat_nomor = plat_nomor  #plat nomor kendaraan
        self.merk = merk  #merk kendaraan
        self.tahun_produksi = tahun_produksi  #tahun produksi kendaraan
        self.warna = warna  #warna kendaraan

    def set_plat_nomor(self, plat_nomor):  #metode untuk mengatur plat nomor kendaraan
        self.plat_nomor = plat_nomor

    def set_merk(self, merk):  #metode untuk mengatur merk kendaraan
        self.merk = merk

    def set_tahun_produksi(self, tahun_produksi):  #metode untuk mengatur tahun produksi kendaraan
        self.tahun_produksi = tahun_produksi

    def set_warna(self, warna):  #metode untuk mengatur warna kendaraan
        self.warna = warna

    def get_plat_nomor(self):  #metode untuk mendapatkan plat nomor kendaraan
        return self.plat_nomor

    def get_merk(self):  #metode untuk mendapatkan merk kendaraan
        return self.merk

    def get_tahun_produksi(self):  #metode untuk mendapatkan tahun produksi kendaraan
        return self.tahun_produksi

    def get_warna(self):  #metode untuk mendapatkan warna kendaraan
        return self.warna

    def display(self):  #metode untuk menampilkan informasi kendaraan
        print("Plat Nomor:", self.plat_nomor)  #tampilkan plat nomor
        print("Merk:", self.merk)  #tampilkan merk
        print("Tahun Produksi:", self.tahun_produksi)  #tampilkan tahun produksi
        print("Warna:", self.warna)  #tampilkan warna
